Move non_differentiable_arg_names from autograd functions to differentiability_info.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #19431 Get rid of unnecessary matches_jit_signature: True specifications.
* #19430 Make one_hot non-differentiable.
* #19429 Remove 'BoolTensor', 'IndexTensor' from frontend specifications.
* #19428 Have _embedding_bag_dense_backward match JIT signature.
* #19427 Have embedding_dense_backward match JIT signature.
* #19426 Hook up non_differentiability in derivatives.yaml when no autograd function is generated.
* **#19425 Move non_differentiable_arg_names from autograd functions to differentiability_info.**
* #19424 Stop generating autograd functions for derivatives.yaml entries that only specify output differentiability.
* #19272 Rename 'not_differentiable' to 'non_differentiable'.

There was a similar problem with as with output_differentiability: if an autograd function is not generated, we just throw away this information.
This doesn't particularly matter in this case, because no codegen depends on it, I'm just moving all argument differentiability arguments off of the autograd function.

Differential Revision: [D15003384](https://our.internmc.facebook.com/intern/diff/D15003384)

gchanan committed:
Move non_differentiable_arg_names from autograd functions to differentiability_info.

There was a similar problem with as with output_differentiability: if an autograd function is not generated, we just throw away this information.
This doesn't particularly matter in this case, because no codegen depends on it, I'm just moving all argument differentiability arguments off of the autograd function.

cpuhrsch commented:
This means Declarations.yaml continues to grow, right? There's a lot of stuff in there now.

gchanan committed:
Update on "Move non_differentiable_arg_names from autograd functions to differentiability_info."

Move non_differentiable_arg_names from autograd functions to differentiability_info.

There was a similar problem with as with output_differentiability: if an autograd function is not generated, we just throw away this information.
This doesn't particularly matter in this case, because no codegen depends on it, I'm just moving all argument differentiability arguments off of the autograd function.

gh-metadata: pytorch pytorch 19425 gh/gchanan/18/head

gchanan committed:
Update on "Move non_differentiable_arg_names from autograd functions to differentiability_info."

Move non_differentiable_arg_names from autograd functions to differentiability_info.

There was a similar problem with as with output_differentiability: if an autograd function is not generated, we just throw away this information.
This doesn't particularly matter in this case, because no codegen depends on it, I'm just moving all argument differentiability arguments off of the autograd function.

gh-metadata: pytorch pytorch 19425 gh/gchanan/18/head

gchanan commented:
> This means Declarations.yaml continues to grow, right? There's a lot of stuff in there now.

No, derivative information is not listed in Declarations.yaml.

gchanan commented:
@pytorchbot retest this please.
